src/core/utils/compatibility.js

```diff
@@ -166,6 +166,13 @@ export function isCompatibleWithUserAgent({
     return { compatible: false, reason: INCOMPATIBLE_NOT_FIREFOX };
   }
 
+  // Fenix does not support add-ons (yet?).
+  // See: https://github.com/mozilla-mobile/fenix/issues/1134
+  // See also: https://github.com/mozilla/addons-frontend/issues/7963
+  if (isFirefoxForAndroid(userAgentInfo)) {
+    return { compatible: false, reason: INCOMPATIBLE_ANDROID_UNSUPPORTED };
```

Comment:
so this replaces the more nuanced logic below that checks the OS and the compatibility by just making all add-ons incompatible with Fennec on Android regardless of the compatibility status?

Reply:
That is correct. For now, we want all add-ons to appear incompatible if someone is running Firefox for Android. There's an issue to change this, when they do become installable, at which point we can being some of the logic back.

Reply:
couldn't we use a feature flag instead then?

Reply:
Yeah, I can probably get that done in time for the tag. This was an attempt at a quick fix for a bug that is currently present in production. I didn't know if we were going to have to hotfix it or not, so I just wanted a patch that fixes it.

Now that I know we can wait until Thursday for this fix, we can probably address #9864 in this patch as well.